src/ledger/NetworkConfig.cpp

```diff
@@ -0,0 +1,475 @@
+// Copyright 2023 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "ledger/NetworkConfig.h"
+#include "util/ProtocolVersion.h"
+#include <optional>
+
+namespace stellar
+{
+namespace
+{
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+void
+createConfigSettingEntry(
+    ConfigSettingEntry const& configSetting, AbstractLedgerTxn& ltxRoot,
+    std::optional<uint32_t> mockLastModifiedLedgerSeqForTests = std::nullopt)
+{
+    LedgerEntry e;
+    e.data.type(CONFIG_SETTING);
+    e.data.configSetting() = configSetting;
+    if (mockLastModifiedLedgerSeqForTests)
+    {
+        e.lastModifiedLedgerSeq = *mockLastModifiedLedgerSeqForTests;
```

Comment:
Not sure why you need this: see `LedgerTxn::Impl::commit()` that will write over `lastModifiedLedgerSeq` using `ledgerSeq`